### PR TITLE
Parsing

### PR DESCRIPTION
**Pull request by lpaffi**
I've fixed the errors you pointed out for the most part.
I added a unparseMethod() which has the sole purpose of translating an int into a enum Method.

I don't know if you can still see the merge request ( I closed it, sorry :p) but I did not implement the switch you asked because switchs only work with integer.

I did not add unitary tests since they're not in the provided file, but we could create our own if you want.

I ran into a problem that I'm not sure how to solve and I'd like your help : when reading the timestamp, we have an additional info, the +0200. There's no such attribute with the mktime, c_time methods, so I'm not sure how to deal with it.

### CODE REVIEW

**Comment by Sadikk** [comment]
The +0200 corresponds to the timezone. It's the %z parameter in strftime (see here : http://www.cplusplus.com/reference/ctime/strftime/ )

Not sure if you're done or not ... ?

Add me as a reviewer (right sidebar on github) when you're done (and you as an assignee) , will then review your code and you will be able to merge if all is okay

**Comment by lpaffi** [comment]
Thanks. I was using %Z and not %z!
I think I'm done with this class =)

File: src/Request/Request.cpp [hunk 1]
```diff
@@ -21,77 +21,107 @@
...
+  else if(unparsedMethod == "PATCH"){
+    return PATCH;
+  }
+}
+
+string Request::unparseMethod(HttpMethod parsedMethod) const {
+  switch(parsedMethod){
+    case 0 :
+    return "GET";
+    break;
```
Comment (Sadikk) [inline src/Request/Request.cpp]: no need to break after a return

File: src/Request/Request.cpp [hunk 2]
```diff
@@ -21,77 +21,107 @@
...
+  }
+  else if(unparsedMethod == "PATCH"){
+    return PATCH;
+  }
+}
+
+string Request::unparseMethod(HttpMethod parsedMethod) const {
+  switch(parsedMethod){
+    case 0 :
+    return "GET";
```
Comment (Sadikk) [inline src/Request/Request.cpp]:
```suggestion
        return "GET";
```

File: src/Request/Request.cpp [hunk 3]
```diff
@@ -189,15 +247,24 @@ Request::~Request ( )
...
 
-
 //------------------------------------------------------------------ PRIVE
 
 //----------------------------------------------------- Méthodes protégées
-void Request::swap(Request& other)
+void Request::swap(Request& first, Request& second)
 {
-  
+std::swap(first.ipAddress, second.ipAddress);
```
Comment (Sadikk) [inline src/Request/Request.cpp:250]: indentation

File: src/Request/Request.h [hunk 4]
```diff
@@ -88,7 +95,7 @@ class Request
 
 protected:
 //----------------------------------------------------- Méthodes protégées
-    void swap(Request& other);
+    void swap(Request& first, Request& second);
```
Comment (Sadikk) [inline src/Request/Request.h]:
```suggestion
    friend void swap(Request& first, Request& second);
```

File: src/Request/Request.h [hunk 5]
```diff
@@ -17,8 +17,8 @@
 
 //------------------------------------------------------------------ Types
 typedef std::string string;
-
-enum HttpMethod {GET, POST, OPTIONS};
+typedef struct tm tm;
+enum HttpMethod {GET, POST, OPTIONS , HEAD, PUT, DELETE, CONNECT, TRACE, PATCH};
 //------------------------------------------------------------------------
 // Rôle de la classe <Request>
 //
```
Comment (Sadikk) [inline src/Request/Request.h]: missing doc

**Review by Sadikk — CHANGES_REQUESTED** [review]
small things to fix ; but you can merge afterwise, it's okay 👍

File: src/Request/Request.h (hunk 4 above)
Comment (lpaffi) [reply src/Request/Request.h]: Fixed

File: src/Request/Request.cpp [hunk 6]
```diff
@@ -94,32 +94,28 @@ HttpMethod Request::parseMethod(string unparsedMethod) const {
 string Request::unparseMethod(HttpMethod parsedMethod) const {
   switch(parsedMethod){
     case 0 :
+<<<<<<< HEAD
```
Comment (Sadikk) [inline src/Request/Request.cpp]: erreur de fusion

File: src/Request/Request.cpp [hunk 7]
```diff
@@ -94,32 +94,28 @@ HttpMethod Request::parseMethod(string unparsedMethod) const {
...
+      return "GET";
+=======
         return "GET";
     break;
+>>>>>>> e3476c8b2ecb0799f8167e93a07fe52c510e8641
```
Comment (Sadikk) [inline src/Request/Request.cpp]: erreur de fusion

**Review by Sadikk — CHANGES_REQUESTED** [review]

**Review by Sadikk — APPROVED** [review]

**Pull request merged**